### DISCUSSION TITLE
openid-client v8 dependency

### DISCUSSION THREAD

**Discussion by HelpfulScripts** [nextauthjs/next-auth#3469] [Help] [answered]
Hi guys,

I am updating to version 4 and ran across the following error when starting my app:
```
error - ./node_modules/openid-client/lib/helpers/deep_clone.js:1:33
Module not found: Can't resolve 'v8'

Import trace for requested module:
./node_modules/openid-client/lib/issuer.js
./node_modules/openid-client/lib/index.js
./node_modules/next-auth/core/lib/oauth/callback.js
./node_modules/next-auth/core/routes/callback.js
./node_modules/next-auth/core/routes/index.js
./node_modules/next-auth/core/index.js
./node_modules/next-auth/next/index.js
./node_modules/next-auth/index.js
./pages/_app.js
```

I don't currently have `v8` installed and am wary to do so because I'm only using simple `Credential` authorization, no `OAuth`. At first glance it seems that `next-auth/core/lib/oauth/callback.js` is only importing 2 type declarations from `openid-client`, which seems to trigger a `require('v8')` call on module intialization.
Is there a workaround that doesn't involve installing `v8`?
Many thanks!

**Comment by balazsorban44** [comment, marked as answer]
I see you are cross posting from https://github.com/panva/node-openid-client/discussions/442


Could you please share your code and explain what you are trying to do?

I'm not really sure how you would end up with this as an error. as explained on the other discussion, v8 is a standard in Node.

**Comment by HelpfulScripts** [comment]
@balazsorban44 thanks for the response. @panva was right, I was using next-auth wrong. The error threw me off a bit, but I got it fixed now.

**Reply by thgh** [reply]
I'm having the same issue. How did you fix it?

**Reply by thgh** [reply]
I think this solved it:
```diff
-import 'next-auth'
+import NextAuth from 'next-auth'
```

**Comment by bboyz269** [comment]
For other dummies like me that hit this page, my "[...nextauth].ts" was in the wrong dir.
`pages/auth/[...nextauth].ts` instead of   
`pages/api/auth/[...nextauth].ts`
